### PR TITLE
Add a dialyxir build to Travis

### PR DESCRIPTION
**Pull request by JayH5**
* PLT files are cached between builds.
  * A build without a cache or with a new Elixir or Erlang version will take about 10 minutes to build the PLT files.
  * A build with a cache takes <2 minutes
* Fixes a few dialyzer warnings
* Update to Dialyxir 1.0.0 RC

### CODE REVIEW

**Comment by JayH5** [comment]
Looks like Travis is failing with a transient test failure in one of the test jobs. Should probably be looked into but I'm not sure where to start 🤔

**Comment by JayH5** [comment]
Build failing due to #87.

File: lib/grpc/adapter/cowboy.ex [hunk 1]
```diff
@@ -177,14 +177,7 @@ defmodule GRPC.Adapter.Cowboy do
...
 
-    addr_str =
-      case addr do
-        :local ->
-          port
-
-        addr ->
-          "#{:inet.ntoa(addr)}:#{port}"
-      end
+    addr_str = "#{:inet.ntoa(addr)}:#{port}"
```
[the remark this answers is not in the document]
Comment (JayH5) [reply lib/grpc/adapter/cowboy.ex]: Yes, you're right, it shouldn't change. The typespec in Ranch seems to not account for Unix domain sockets, so Dialyzer thinks this is an error. I created an issue upstream: ninenines/ranch#213.

**Comment by tony612** [comment]
Closing by https://github.com/elixir-grpc/grpc/pull/127

**Pull request closed without merging**